Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -0,0 +1,158 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Globalization;
+    using System.IO;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A store for instrumentation App Ids. This makes sure we don't query the public endpoint to find an app Id for the same Ikey more than once.
+    /// </summary>
+    internal static class CorrelationIdLookupHelper
```

Comment:
why do you want this class to be static? Why wouldn't modules have an instace of it that they can pre-initialize?